pkg/appregistry/client.go

```diff
@@ -6,6 +6,17 @@ import (
 	"strings"
 )
 
+// Client exposes the functionality of app registry server
+type Client interface {
+	// RetrieveAll retrieves all visible packages from the given source
+	// When namespace is specified, only package(s) associated with the given namespace are returned.
+	// If namespace is empty then visible package(s) across all namespaces are returned.
+	RetrieveAll(namespace string) ([]*OperatorMetadata, error)
```

Comment:
Like I mentioned above it is cleaner to also have `RetrieveAll()`. You can either add that as part of this PR or open a new PR for that.